### PR TITLE
keep-presence: init at 1.0.5

### PR DESCRIPTION
**Pull request by mwdomino**
This is my first contribution to `nixpkgs` so please let me know if I've missed anything!

###### Description of changes
This adds a new package `keep-presence` from https://github.com/carrot69/keep-presence.

keep-presence is a simple python script to move your mouse and keep your computer active. Can be used to prevent sleep or to keep yourself 'active' on the current chat app of choice.

~~I'll be opening a PR to the repo above to see if the maintainer is interested in integrating the changes. If so, I will update this package at that time to use the primary repo instead of my fork.~~ The original author has merged the PR so I've updated this package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/keep-presence/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{ nixpkgs ? import <nixpkgs> {}, pythonPkgs ? nixpkgs.pkgs.python3Packages }:
```
Comment (ehmry) [inline pkgs/tools/misc/keep-presence/default.nix:1]: Nixpkgs is never imported from within nixpkgs, just make this file a function that takes its inputs as arguments and `callPackage` will do the rest.

File: pkgs/tools/misc/keep-presence/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+let
+  inherit (nixpkgs) pkgs lib fetchFromGitHub;
+  inherit pythonPkgs;
+
+  f = { buildPythonPackage, setuptools, pynput }:
+    buildPythonPackage rec {
```
Comment (ehmry) [inline pkgs/tools/misc/keep-presence/default.nix:8]: `buildPythonApplication`?

File: pkgs/tools/misc/keep-presence/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        sha256 = "1a4r35pbdmd73767x3g41c3chjxjwqf3y4n3dhmh31s1lx7cahmv";
+      } + /src;
+
+      propagatedBuildInputs = with pythonPkgs; [ setuptools pynput ];
+
+      doCheck = false;
+
+      postInstall = ''
+        mv -v $out/bin/keep-presence.py $out/bin/keep-presence
+        patchShebangs $out/bin/keep-presence
```
Comment (ehmry) [inline pkgs/tools/misc/keep-presence/default.nix:39]: I would expect shebangs to be taken care automatically, but I could be wrong.

File: pkgs/tools/misc/keep-presence/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        chmod +x $out/bin/keep-presence
+      '';
+    };
+
+  drv = pythonPkgs.callPackage f {};
```
Comment (ehmry) [inline pkgs/tools/misc/keep-presence/default.nix:44]: This file should only contain `f`.

**Review by ghost — CHANGES_REQUESTED** [review]
It is preferable to make changes to commiters and changes to packages in seperate commits.

**Comment by Janik-Haag** [comment]
@mwdomino are you still interested in getting this merged?

**Comment by abathur** [comment]
Tentatively closing this since the requested changes haven't been made and Janik's query has gone unanswered for about 2 months.

@mwdomino Please don't let this discourage you if you want to get this merged. I'm more than happy to reopen it if you are :)

**Pull request closed without merging**